homeassistant/components/maxcube/climate.py

```diff
@@ -42,6 +42,20 @@
 
 SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
 
+HASS_PRESET_TO_MAX_MODE = {
+    PRESET_AWAY: MAX_DEVICE_MODE_VACATION,
+    PRESET_BOOST: MAX_DEVICE_MODE_BOOST,
+    PRESET_NONE: MAX_DEVICE_MODE_AUTOMATIC,
+    PRESET_ON: MAX_DEVICE_MODE_MANUAL,
```

Comment:
Should it be `PRESET_ON` or `PRESET_NONE`?

Reply:
What should it be then? I'm asking since this isn't mapped the same way in the other dict below.

Reply:
The problem is that there's no one-to-one mapping between MAX! Cube modes and HA HVAC modes. Proper mapping depends on additional data. Here's a table of how it looks:

| MAX! Cube UI | MAX! Cube Mode | MAX! Cube target t°C | HA HVAC mode | HA preset |
| --- | --- | --- | --- | --- |
| Auto (schedule) | `MAX_DEVICE_MODE_AUTOMATIC` | *any* | `HVAC_MODE_AUTO` | `PRESET_NONE` |
| Boost | `MAX_DEVICE_MODE_BOOST` | *any* | `HVAC_MODE_AUTO` | `PRESET_BOOST` |
| Vacation | `MAX_DEVICE_MODE_VACATION` | *any* | `HVAC_MODE_MANUAL` | `PRESET_AWAY` |
| Eco | `MAX_DEVICE_MODE_MANUAL`¹ | User configured Eco² | `HVAC_MODE_MANUAL` | `PRESET_ECO` |
| Comfort | `MAX_DEVICE_MODE_MANUAL`¹ | User configured Comfort² | `HVAC_MODE_MANUAL` | `PRESET_COMFORT` |
| On | `MAX_DEVICE_MODE_MANUAL`¹ | 30.5 | `HVAC_MODE_MANUAL` | `PRESET_ON` |
| Off | `MAX_DEVICE_MODE_MANUAL`¹ | 4.5 | `HVAC_MODE_OFF` | `PRESET_NONE` |

¹ - also `MAX_DEVICE_MODE_AUTOMATIC`, but we ignore that case because schedule will override it eventually
² - any value between 5.0 and 30.0 (inclusive), stored on device when it's configured

This dictionary is just a helper and proper mapping is handled inside `preset_mode()` and `set_preset_mode()`.

Reply:
Ok.